File: stipend_sign_flutter/lib/screens/register_screen.dart
import 'package:flutter/material.dart';
import '../services/storage_service.dart';
import '../models/user.dart';
import 'profile_screen.dart';

class RegisterScreen extends StatefulWidget {
  const RegisterScreen({super.key});

  @override
  State<RegisterScreen> createState() => _RegisterScreenState();
}

class _RegisterScreenState extends State<RegisterScreen> {
  final _emailCtrl = TextEditingController();
  final _passwordCtrl = TextEditingController();
  final _confirmCtrl = TextEditingController();
  final _firstNameCtrl = TextEditingController();
  final _lastNameCtrl = TextEditingController();
  final _studentIdCtrl = TextEditingController();
  final _departmentCtrl = TextEditingController();
  final _yearCtrl = TextEditingController();
  final _phoneCtrl = TextEditingController();
  bool _loading = false;
  bool _obscure = true;

  @override
  void dispose() {
    for (final c in [
      _emailCtrl, _passwordCtrl, _confirmCtrl, _firstNameCtrl,
      _lastNameCtrl, _studentIdCtrl, _departmentCtrl, _yearCtrl, _phoneCtrl
    ]) {
      c.dispose();
    }
    super.dispose();
  }

  Future<void> _handleRegister() async {
    if (_emailCtrl.text.trim().isEmpty ||
        _passwordCtrl.text.isEmpty ||
        _firstNameCtrl.text.trim().isEmpty ||
        _lastNameCtrl.text.trim().isEmpty ||
        _studentIdCtrl.text.trim().isEmpty) {
      _showError('Please fill in all required fields');
      return;
    }

    if (_passwordCtrl.text != _confirmCtrl.text) {
      _showError('Passwords do not match');
      return;
    }

    if (_passwordCtrl.text.length < 6) {
      _showError('Password must be at least 6 characters');
      return;
    }

    setState(() => _loading = true);

    try {
      final result = await StorageService.registerUser(
        email: _emailCtrl.text.trim(),
        password: _passwordCtrl.text,
        firstName: _firstNameCtrl.text.trim(),
        lastName: _lastNameCtrl.text.trim(),
        studentId: _studentIdCtrl.text.trim(),
        department: _departmentCtrl.text.trim(),
        yearOfStudy: _yearCtrl.text.trim(),
        phone: _phoneCtrl.text.trim(),
      );

      if (result['success'] == true) {
        final user = result['user'] as User;
        if (mounted) {
          Navigator.of(context).pushReplacement(
            MaterialPageRoute(builder: (_) => ProfileScreen(user: user)),
          );
        }
      } else {
        _showError(result['message'] as String);
      }
    } finally {
      if (mounted) setState(() => _loading = false);
    }
  }

  void _showError(String msg) {
    ScaffoldMessenger.of(context).showSnackBar(
      SnackBar(content: Text(msg), backgroundColor: Colors.red),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Create Account')),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(20),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            _sectionTitle('Account Details'),
            _field(_emailCtrl, 'Email *', TextInputType.emailAddress),
            _field(_passwordCtrl, 'Password *', TextInputType.text, obscure: _obscure, suffixIcon: IconButton(
              icon: Icon(_obscure ? Icons.visibility_off : Icons.visibility),
              onPressed: () => setState(() => _obscure = !_obscure),
            )),
            _field(_confirmCtrl, 'Confirm Password *', TextInputType.text, obscure: _obscure),
            const SizedBox(height: 8),
            _sectionTitle('Personal Information'),
            _field(_firstNameCtrl, 'First Name *', TextInputType.name),
            _field(_lastNameCtrl, 'Last Name *', TextInputType.name),
            _field(_studentIdCtrl, 'Student ID *', TextInputType.text),
            _field(_departmentCtrl, 'Department', TextInputType.text),
            _field(_yearCtrl, 'Year of Study (1-5)', TextInputType.number),
            _field(_phoneCtrl, 'Phone Number', TextInputType.phone),
            const SizedBox(height: 16),
            SizedBox(
              height: 50,
              child: ElevatedButton(
                onPressed: _loading ? null : _handleRegister,
                style: ElevatedButton.styleFrom(
                  backgroundColor: const Color(0xFF4CAF50),
                  foregroundColor: Colors.white,
                  shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
                ),
                child: _loading
                    ? const SizedBox(
                        height: 20,
                        width: 20,
                        child: CircularProgressIndicator(color: Colors.white, strokeWidth: 2),
                      )
                    : const Text('Register', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
              ),
            ),
            TextButton(
              onPressed: () => Navigator.of(context).pop(),
              child: const Text(
                'Already have an account? Login',
                style: TextStyle(color: Color(0xFF4CAF50), fontSize: 16),
              ),
            ),
            const SizedBox(height: 20),
          ],
        ),
      ),
    );
  }

  Widget _sectionTitle(String title) => Padding(
        padding: const EdgeInsets.only(bottom: 12, top: 4),
        child: Text(title,
            style: const TextStyle(
                fontSize: 18, fontWeight: FontWeight.bold, color: Color(0xFF4CAF50))),
      );

  Widget _field(
    TextEditingController ctrl,
    String label,
    TextInputType type, {
    bool obscure = false,
    Widget? suffixIcon,
  }) =>
      Padding(
        padding: const EdgeInsets.only(bottom: 12),
        child: TextField(
          controller: ctrl,
          keyboardType: type,
          obscureText: obscure,
          decoration: InputDecoration(labelText: label, suffixIcon: suffixIcon),
        ),
      );
}

File: sign in stipend app/screens/RegisterScreen.js
// filepath: screens/RegisterScreen.js
import React, { useState } from 'react';
import { View, Text, TextInput, TouchableOpacity, StyleSheet, Alert, ScrollView } from 'react-native';
import { registerUser } from '../utils/Storage';
import BackButton from '../components/BackButton';

export default function RegisterScreen({ navigation }) {
  const [formData, setFormData] = useState({
    email: '',
    password: '',
    confirmPassword: '',
    firstName: '',
    lastName: '',
    studentId: '',
    department: '',
    yearOfStudy: '',
    phone: '',
  });
  const [loading, setLoading] = useState(false);

  const updateField = (field, value) => {
    setFormData(prev => ({ ...prev, [field]: value }));
  };

  const handleRegister = async () => {
    // Validate required fields
    if (!formData.email || !formData.password || !formData.studentId || 
        !formData.firstName || !formData.lastName) {
      Alert.alert('Error', 'Please fill in all required fields');
      return;
    }

    if (formData.password !== formData.confirmPassword) {
      Alert.alert('Error', 'Passwords do not match');
      return;
    }

    if (formData.password.length < 6) {
      Alert.alert('Error', 'Password must be at least 6 characters');
      return;
    }

    setLoading(true);
    
    try {
      const result = await registerUser({
        email: formData.email,
        password: formData.password,
        firstName: formData.firstName,
        lastName: formData.lastName,
        studentId: formData.studentId,
        department: formData.department,
        yearOfStudy: formData.yearOfStudy,
        phone: formData.phone,
        studentCardImage: null,
        biometricData: null,
      });
      
      if (result.success) {
        Alert.alert('Success', 'Account created! Please complete your profile.', [
          { text: 'OK', onPress: () => navigation.replace('Profile', { user: result.user }) }
        ]);
      } else {
        Alert.alert('Error', result.message);
      }
    } catch (error) {
      Alert.alert('Error', 'Something went wrong');
    }
    
    setLoading(false);
  };

  return (
    <ScrollView style={styles.container}>
      <BackButton navigation={navigation} />
      <Text style={styles.header}>Create Student Account</Text>
      
      <View style={styles.section}>
        <Text style={styles.sectionTitle}>Account Details</Text>
        
        <TextInput
          style={styles.input}
          placeholder="Email *"
          value={formData.email}
          onChangeText={(v) => updateField('email', v)}
          keyboardType="email-address"
          autoCapitalize="none"
        />
        
        <TextInput
          style={styles.input}
          placeholder="Password *"
          value={formData.password}
          onChangeText={(v) => updateField('password', v)}
          secureTextEntry
        />
        
        <TextInput
          style={styles.input}
          placeholder="Confirm Password *"
          value={formData.confirmPassword}
          onChangeText={(v) => updateField('confirmPassword', v)}
          secureTextEntry
        />
      </View>

      <View style={styles.section}>
        <Text style={styles.sectionTitle}>Personal Information</Text>
        
        <TextInput
          style={styles.input}
          placeholder="First Name *"
          value={formData.firstName}
          onChangeText={(v) => updateField('firstName', v)}
        />
        
        <TextInput
          style={styles.input}
          placeholder="Last Name *"
          value={formData.lastName}
          onChangeText={(v) => updateField('lastName', v)}
        />
        
        <TextInput
          style={styles.input}
          placeholder="Student ID *"
          value={formData.studentId}
          onChangeText={(v) => updateField('studentId', v)}
        />
        
        <TextInput
          style={styles.input}
          placeholder="Department"
          value={formData.department}
          onChangeText={(v) => updateField('department', v)}
        />
        
        <TextInput
          style={styles.input}
          placeholder="Year of Study (1-5)"
          value={formData.yearOfStudy}
          onChangeText={(v) => updateField('yearOfStudy', v)}
          keyboardType="numeric"
        />
        
        <TextInput
          style={styles.input}
          placeholder="Phone Number"
          value={formData.phone}
          onChangeText={(v) => updateField('phone', v)}
          keyboardType="phone-pad"
        />
      </View>

      <TouchableOpacity 
        style={styles.button} 
        onPress={handleRegister}
        disabled={loading}
      >
        <Text style={styles.buttonText}>
          {loading ? 'Creating Account...' : 'Register'}
        </Text>
      </TouchableOpacity>

      <TouchableOpacity 
        style={styles.linkButton}
        onPress={() => navigation.goBack()}
      >
        <Text style={styles.linkText}>Already have an account? Login</Text>
      </TouchableOpacity>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    padding: 20,
  },
  header: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#333',
    marginBottom: 20,
    textAlign: 'center',
  },
  section: {
    marginBottom: 20,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#4CAF50',
    marginBottom: 15,
  },
  input: {
    borderWidth: 1,
    borderColor: '#ddd',
    borderRadius: 10,
    padding: 15,
    marginBottom: 12,
    fontSize: 16,
  },
  button: {
    backgroundColor: '#4CAF50',
    padding: 15,
    borderRadius: 10,
    alignItems: 'center',
    marginTop: 10,
  },
  buttonText: {
    color: '#fff',
    fontSize: 18,
    fontWeight: 'bold',
  },
  linkButton: {
    marginTop: 20,
    marginBottom: 40,
    alignItems: 'center',
  },
  linkText: {
    color: '#4CAF50',
    fontSize: 16,
  },
});